rct229/rulesets/ashrae9012019/section19/section19rule8.py

```diff
@@ -0,0 +1,150 @@
+                    zone_id_b = zone_b["id"]
+                    zone_space_b[zone_id_b] = []
+                    for space_b in getattr_(zone_b, "zone", "spaces"):
+                        zone_space_b[zone_id_b].append(space_b)
+
+                return {"zone_space_b": zone_space_b}
+
+            def list_filter(self, context_item, data=None):
+                hvac_b = context_item.baseline
+                hvac_id_b = hvac_b["id"]
+                hvac_zone_list_w_area_dict_b = data["hvac_zone_list_w_area_dict_b"]
+
+                return True if hvac_id_b in hvac_zone_list_w_area_dict_b else False
+
+            class HVACRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section19Rule8.BuildingRule.BuildingSegmentRule.HVACRule, self
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        required_fields={
+                            "$": ["fan_system"],
+                            "fan_system": ["minimum_outdoor_airflow"],
+                        },
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    hvac_b = context.baseline
+                    hvac_zone_list_w_area_dict_b = data["hvac_zone_list_w_area_dict_b"]
+                    zone_space_b = data["zone_space_b"]
+
+                    hvac_id_b = hvac_b["id"]
+                    fan_system_b = hvac_b["fan_system"]
+
+                    is_DCV_modeled_b = False
+                    avg_occ_density = 0.0 / ureg("ft2")
+                    hvac_min_OA_flow = fan_system_b["minimum_outdoor_airflow"]
+
+                    demand_control_ventilation_control_b = fan_system_b.get(
+                        "demand_control_ventilation_control"
+                    )
+
+                    if (
+                        demand_control_ventilation_control_b
+                        and demand_control_ventilation_control_b
+                        != DEMAND_CONTROL_VENTILATION_CONTROL.NONE
+                    ):
+                        is_DCV_modeled_b = True
+                        if hvac_min_OA_flow > MIN_OA_CFM:
+                            zone_list_b = hvac_zone_list_w_area_dict_b[hvac_id_b][
+                                "zone_list"
+                            ]
+                            hvac_area_b = hvac_zone_list_w_area_dict_b[hvac_id_b][
+                                "total_area"
+                            ]
+                            total_hvac_sys_occupants_b = 0.0
+                            for zone_b in zone_list_b:
+                                for space_b in zone_space_b[zone_b]:
+                                    total_hvac_sys_occupants_b += space_b[
+                                        "number_of_occupants"
+                                    ]
+
+                            avg_occ_density = total_hvac_sys_occupants_b / hvac_area_b
+
```

Comment:
With above modifications, this can be simplified into
```python
avg_occ_density = sum([zone_space_b[zone_id_b] for zone_id_b in zone_id_list_b]) / hvac_area_b
```
Also, you can remove line 122 - 127

Reply:
Addressed.